src/sql/src/pure.rs

```diff
@@ -143,6 +146,56 @@ fn subsource_name_gen(
     Ok(UnresolvedItemName::from(partial))
 }
 
+//// Validates the requested subsources do not have name conflicts with each other
+/// and that the same upstream table is not referenced multiple times.
+fn validate_subsource_names<T>(
```

Comment:
this is pure code-movement from `validate_requested_subsources` in `src/sql/src/pure/postgres.rs` so it can be re-used for mysql.

Reply:
Amazing! :tada: